src/main/kotlin/build/buf/intellij/annotator/BufAnalyzeUtils.kt

```diff
@@ -185,12 +191,25 @@ object BufAnalyzeUtils {
         )
         handler.addProcessListener(object : ProcessAdapter() {
             override fun onTextAvailable(event: ProcessEvent, outputType: com.intellij.openapi.util.Key<*>) {
-                result.add(event.text.trimEnd())
+                when (outputType) {
+                    ProcessOutputType.STDOUT -> stdout.add(event.text.trimEnd())
+                }
+            }
+
+            override fun processTerminated(event: ProcessEvent) {
+                exitCode.set(event.exitCode)
             }
         }, owner)
         handler.startNotify()
-        handler.waitFor(BUF_COMMAND_EXECUTION_TIMEOUT.toMillis())
-        result.drop(1) // first line is always the CMD executed
+        if (handler.waitFor(BUF_COMMAND_EXECUTION_TIMEOUT.toMillis())) {
+            if (exitCode.get() != 0) {
+                LOG.warn("buf exit code: $exitCode")
+            }
+        } else {
+            // Process failed to complete within given timeout - stop it
+            handler.destroyProcess()
```

Comment:
If the buf CLI process didn't run to completion in 2m, we should force it to stop.